src/program.ts

```diff
@@ -3149,6 +3192,28 @@ export class Local extends VariableLikeElement {
     assert(type != Type.void);
     this.setType(type);
   }
+
+  close(offset: usize): ClosedLocal {
+    return new ClosedLocal(this, offset);
+  }
+}
+
+/** a closed-over local variable. */
+export class ClosedLocal extends VariableLikeElement {
```

Comment:
Having a separate class here makes me wonder if we lose the ability to record local state for closed vars in flows. Like, flows observe whether a small integer local needs to be wrapped on the next operation requiring it or can remain unwrapped to avoid unnecessary operations, or whether it is known-to-be retained and stuff like that. What happens there with `ClosedLocal`s?

Reply:
Could probably just get rid of it and just make a property on local, then it shouldn't be a problem